packages/next/client/experimental-script.tsx

```diff
@@ -0,0 +1,169 @@
+import React, { useEffect, useContext } from 'react'
+import { ScriptHTMLAttributes } from 'react'
+import { HeadManagerContext } from '../next-server/lib/head-manager-context'
+
+const ScriptCache = new Map()
+const LoadCache = new Set()
+
+const DOMAttributeNames: Record<string, string> = {
+  acceptCharset: 'accept-charset',
+  className: 'class',
+  htmlFor: 'for',
+  httpEquiv: 'http-equiv',
+}
+
+interface Props extends ScriptHTMLAttributes<HTMLScriptElement> {
+  strategy?: 'defer' | 'lazy' | 'dangerousRenderBlocking' | 'eager'
+  key?: string
+  onLoad?: () => void
+  onError?: () => void
+  children?: any
+  preload?: boolean
```

Comment:
Why is preload a boolean? Wouldn't this be controlled by the selected strategy?

Reply:
This is an extra option mainly to experiment adding preload with `defer` strategy. The idea being test load and execution separately. Otherwise the preloads are controlled